lib/matplotlib/gridspec.py

```diff
@@ -478,6 +483,18 @@ def get_rows_columns(self):
         row_stop, col_stop = divmod(self.num2, ncols)
         return nrows, ncols, row_start, row_stop, col_start, col_stop
 
+    @property
+    def rowspan(self):
+        """The rows spanned by this subplot, as a `slice` object."""
+        _, _, row_start, row_stop, _, _ = self.get_rows_columns()
+        return slice(row_start, row_stop + 1)
+
+    @property
+    def colspan(self):
+        """The columns spanned by this subplot, as a `slice` object."""
+        _, _, _, _, col_start, col_stop = self.get_rows_columns()
+        return slice(col_start, col_stop + 1)
```

Comment:
`get_rows_columns` was just written as a helper (prob should have been private?, sorry!)  Does it make more sense for this slice just to be an actual attribute that gets set at init, and `get_rows_columns` just uses that?

Reply:
The problem is that it is explicitly allowed to change a subplotspec's position in the grid by assigning to it (there's even a test for that!), and I wouldn't be surprised if it was possible to change the parent gridspec's geometry as well, so then you need to keep both num1/num2 and rowspan/colspan in sync, which just sounds overkill (it's not as if recomputing the rowspan and colspan every time was going to be a bottleneck...).

Reply:
Oh good lord, someone overthought that!

Reply:
I changed these 1) to not use get_rows_columns() as I'll just deprecate it in a later PR (especially if you suggest it should have been private :)) in favor of using rowspan/colspan, and 2) to return a range object rather than a slice object (the difference is small, but basically the range always has start and stop as integers (whereas a slice could have start or stop = None (corresponding to `obj[:stop]` or `obj[start:]`) or use negative values to mean "from the end", even though here I did normalize them to positive integers); plus, ranges support `x in range` and are hashable if we ever need that).

Reply:
That all seems reasonable to me.